packages/git-tools/scripts/git-valu-assert-sync

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+set -eu
+set -o pipefail
+
+git fetch origin --quiet
+branch="$(git rev-parse --abbrev-ref HEAD)"
+git push origin $branch:$branch
+if [ "$(git rev-parse HEAD)" != "$(git rev-parse origin/$branch)" ]; then
+  echo "You local git is not sync with the remote origin/$branch"
+  echo "Ensure it is synced with: git pull origin && git push origin"
+  exit 1
+fi
```

Comment:
Looks off. After push it is alway sync and the assert cannot ever fail

I'd consider adding a dirty check here too since uncommitted changes cannot be in "sync".

```suggestion

if [ "$(git status --porcelain)" != "" ]; then
  echo "Dirty git."
  exit 1
fi

if [ "$(git rev-parse HEAD)" != "$(git rev-parse origin/$branch)" ]; then
  echo "You local git is not sync with the remote origin/$branch"
  echo "Ensure it is synced with: git pull origin && git push origin"
  exit 1
fi
```

Reply:
also mutating the origin in an "assert" is bit surprising